ml/src/main/scala/org/apache/spark/ml/spark/models/svm/SVM.java

```diff
@@ -188,12 +192,19 @@ public void computeImpl() {
             svm.optimizer().setGradient(_parms._gradient.get());
             svm.optimizer().setUpdater(_parms._updater.get());
 
+            ProgressListener progressBar = new ProgressListener(_job, RDDInfo.fromRdd(training), asScalaIterable(Arrays.<String>asList("treeAggregate")));
```

Comment:
Unfortunately no, Spark gets the name dynamically from code so the whole name is `treeAggregate at GradientDescent:227` so from which file, which line and as what method was the job submitted. That also why it's going to be hard to track their changes in code should they change the name of the method.

The only other way I figured out is not to check the `stageName` in the listener and just progress the bar on each `jobStart` for a given RDD but then the number of such event will be bigger than the total number set in SVM algo (and there's no way to check how many jobs will be submitted by spark beforehand).